### PR TITLE
Make player rotate with movement

### PR DESCRIPTION
**Pull request by LordDeatHunter**
thetawave_7RH5OFXIyc
thetawave_bjNxzli7OC

### CODE REVIEW

**Review by cdsupina — APPROVED** [review]
This looks good to me! I'll let @varoonp123 take a look before we merge.

**Comment by LordDeatHunter** [comment]
Seems like with the movement+rotation separation, it applies the rotation to everything. It looks cool on some enemies, but there are issues with the rockets (not homing anymore) and the walls (their rotation defaults to horizontal).
Should I make the rotation player-specific, or should I make a new "tiltable" component?

**Comment by cdsupina** [comment]
> Seems like with the movement+rotation separation, it applies the rotation to everything. It looks cool on some enemies, but there are issues with the rockets (not homing anymore) and the walls (their rotation defaults to horizontal). Should I make the rotation player-specific, or should I make a new "tiltable" component?

I like the idea of making a it a component. There are certainly some mobs where it wouldn't make much sense.

**Review by cdsupina — APPROVED** [review]

**Pull request merged**
